src/EditModel/event-program/tracker-program/epics.js

```diff
@@ -132,34 +157,31 @@ export const cancelProgramStageEdit = action$ =>
         .ofType(PROGRAM_STAGE_EDIT_CANCEL)
         .flatMap(() =>
             programStore.take(1).map(store => {
-                const program = store.program;
-                const stageId = store.programStageToEditCopy.id;
-                const index = getProgramStageIndexById(stageId)(store);
+                try {
+                    const stageId = store.programStageToEditCopy.id;
+                    const index = getProgramStageIndexById(stageId)(store);
 
-                if (index < 0) {
-                    console.warn(
-                        `ProgramStage with id ${stageId} does not exist`
-                    );
-                }
-                let model = store.programStageToEditCopy;
-                let programStageSetter = set(
-                    `programStages[${index}]`,
-                    model,
-                    store
-                );
-                // If the programstage is new, remove it when cancelling
-                if (store.programStageToEditCopy.name === undefined) {
-                    const removedFromProgramStages = store.programStages.filter(
-                        (p, i) => i !== index
-                    );
-                    program.programStages.remove(stageId);
-                    programStageSetter = set(
-                        'programStages',
-                        removedFromProgramStages,
+                    if (index < 0) {
+                        console.warn(
```

Comment:
You use `console` to log in this file and `loglevel` in the next. I think `console` is fine for development, but we should probably use `loglevel` for logs we want to ship to clients.